extensions/amp-viewer-integration/0.1/messaging/messaging.js

```diff
@@ -392,3 +391,39 @@ export class Messaging {
     return err ? (err.message ? err.message : String(err)) : 'unknown error';
   }
 }
+
+/**
+ * Performs a handshake and initializes messaging.
+ * @param {!Window} source
+ * @param {!HTMLIFrameElement} iframe
+ * @param {string} origin
+ * @return {!Promise<!Messaging>}
+ */
+export function initMessaging(source, iframe, origin) {
+  const target = iframe.contentWindow;
+  return new Promise(resolve => {
+    const listener = e => {
+      const data = e.data;
+      if (
+        e.origin === origin &&
+        e.source === target &&
+        data['app'] === APP &&
+        data['name'] === CHANNEL_OPEN_MSG
+      ) {
+        source.removeEventListener('message', listener);
+
+        const port = new WindowPortEmulator(source, origin, target);
+        port.postMessage(
+          /** @type {!AmpViewerMessage} */ ({
```

Comment:
Hmm. This typing doesn't really make sense to me. But I assume it comes from the old code?

Reply:
I saw it used above for `sendMessage_` so I added it here as well. Should I remove it?

Reply:
Let's keep for now and see if Travis fails. And we need to get @erwinmombay 's opinion on what to do with this in general since this is not a true typedef - this appears to require just a JsonObject.